lib/improver/cli/apply_emos_coefficients.py

```diff
@@ -0,0 +1,202 @@
+
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.argparser import ArgParser
+from improver.ensemble_calibration.ensemble_calibration import (
+    ApplyCoefficientsFromEnsembleCalibration)
+from improver.ensemble_copula_coupling.ensemble_copula_coupling import (
+    EnsembleReordering,
+    GeneratePercentilesFromMeanAndVariance,
+    GeneratePercentilesFromProbabilities,
+    GenerateProbabilitiesFromMeanAndVariance,
+    RebadgePercentilesAsRealizations,
+    ResamplePercentiles)
+from improver.utilities.cube_checker import find_percentile_coordinate
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main(argv=None):
+    """Load in arguments for applying coefficients for Ensemble Model Output
+       Statistics (EMOS), otherwise known as Non-homogeneous Gaussian
+       Regression (NGR). The coefficients are applied to the forecast
+       that is supplied, so as to calibrate the forecast. The calibrated
+       forecast is written to a netCDF file.
+    """
+    parser = ArgParser(
+        description='Apply coefficients for Ensemble Model Output '
+                    'Statistics (EMOS), otherwise known as Non-homogeneous '
+                    'Gaussian Regression (NGR)')
+    # Filepaths for the forecast, EMOS coefficients and the output.
+    parser.add_argument('forecast_filepath', metavar='FORECAST_FILEPATH',
+                        help='A path to an input NetCDF file containing the '
+                             'forecast to be calibrated.')
+    parser.add_argument('coefficients_filepath',
+                        metavar='COEFFICIENTS_FILEPATH',
+                        help='A path to an input NetCDF file containing the '
+                             'coefficients used for calibration.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILEPATH',
+                        help='The output path for the processed NetCDF')
+    # Optional arguments.
+    parser.add_argument('--num_realizations', metavar='NUMBER_OF_REALIZATIONS',
+                        default=None, type=np.int32,
+                        help='Optional argument to specify the number of '
+                             'ensemble realizations to produce. '
+                             'If the current forecast is input as '
+                             'probabilities or percentiles then this argument '
+                             'is used to create the requested number of '
+                             'realizations. In addition, this argument is '
+                             'used to construct the requested number of '
+                             'realizations from the mean and variance output '
+                             'after applying the EMOS coefficients.'
+                             'Default will be the number of realizations '
+                             'in the raw input file.')
+    parser.add_argument('--random_ordering', default=False,
```

Comment:
Might be worth saying this is only used for realization input.

Reply:
Extended docstring.